app/assets/javascripts/discourse/app/models/site.js

```diff
@@ -110,6 +81,13 @@ export default class Site extends RestModel.extend().reopenClass(Singleton) {
 
   @sort("categories", "topicCountDesc") categoriesByCount;
 
+  @computed("categories.[]")
+  get categoriesById() {
+    const map = new Map();
+    this.categories.forEach((c) => map.set(c.id, c));
+    return map;
+  }
```

Comment:
Searching through core for `categoriesById`, i see quite a few places doing `site.categoriesById[]`. I guess those will be broken now that we're using a real map rather than setting properties?

Reply:
I looked again and I could not find any other place. Is VS Code failing me? Technically, it should only be accessed through `Category.findById`.

Reply:
Looks like mostly tests, and then a couple in some plugins I have checked out locally:
<img width="714" alt="SCR-20240326-mlyj" src="https://github.com/discourse/discourse/assets/6270921/3ec5c8f8-a17f-4b9e-a453-557021c65bbb">

Maybe worth searching all-the-* as well?